admin/spam/views.py

```diff
@@ -1,96 +1,124 @@
 from __future__ import unicode_literals
 
 from django.shortcuts import render
-from django.contrib.auth.decorators import login_required
 from django.core.mail import send_mail
-from django.http import HttpResponseNotFound
 from django.views.generic import FormView, ListView
-from django.utils.decorators import method_decorator
-from django.core.urlresolvers import reverse
 from django.views.defaults import page_not_found
 
 from modularodm import Q
 from website.project.model import Comment
 from website.settings import SUPPORT_EMAIL
 
-from admin.common_auth.logs import update_admin_log, CONFIRM_HAM, CONFIRM_SPAM
+from admin.base.utils import OSFAdmin
+from admin.common_auth.logs import (
+    update_admin_log,
+    CONFIRM_HAM,
+    CONFIRM_SPAM,
+    USER_EMAILED,
+)
 from admin.spam.serializers import serialize_comment
 from admin.spam.forms import EmailForm, ConfirmForm
+from admin.spam.templatetags.spam_extras import reverse_spam_detail
 
+STATUS = dict(
```

Comment:
@Ghalko My fifteen minutes turned into several hours. I just started looking at this now. I'm going to pick it up again first thing in the morning, fresh. My first note would be to take a look at django's [ChoiceField](https://docs.djangoproject.com/en/1.9/ref/forms/fields/#choicefield). You can make a tuple of tuples in your [model](https://docs.djangoproject.com/en/1.9/ref/models/fields/#field-choices) and give choices there. It's the same concept as what I think you're doing but it's bound to the model and by setting this to a ChoiceField on the ModelForm it automatically selects a widget based on whether multiple select is allowed and creates the right html. You can also use it on multiple field types, Integer, Char, whatever. If you want to get really fancy there's [this](https://django-model-utils.readthedocs.org/en/latest/utilities.html#choices). It's part of a third-party package I've used in the past. That package has a ton of other nice features around model managers that cut out a lot of boiler-plate code.